File: robocode2-server/src/main/java/net/robocode2/events/BulletFiredEvent.java
package net.robocode2.events;

import lombok.Value;
import net.robocode2.model.Bullet;

/**
 * Event sent when a bullet has fired a bullet.
 * 
 * @author Flemming N. Larsen
 */
@Value
public class BulletFiredEvent implements Event {

	/** Fired bullet */
	Bullet bullet;
}

File: robocode2-gui/src/main/kotlin/net/robocode2/gui/model/event/BulletFiredEvent.kt
package net.robocode2.gui.model.event

import net.robocode2.gui.model.comm.Content
import net.robocode2.gui.model.comm.ContentType

class BulletFiredEvent(
        val bullet: BulletState
) : Content(type = ContentType.BULLET_FIRED_EVENT.type)
